### Основные этапы обработки временных рядов разделены на несколько уроков :
1. Разведочный анализ.
2. Преобразование признаков для моделей прогнозирования.
3. Селекция признаков применительно к задаче прогнозирования.
4. Построение моделей прогнозирования временного ряда.

Загрузим данные и разделим их на тренировочную и тестовую выборку

In [5]:
%matplotlib inline
import matplotlib.pyplot as plt

import numpy as np 
import pandas as pd 
import seaborn as sns

from sklearn.metrics import mean_squared_error

import os
#print(os.listdir())

In [6]:
import datetime as dt
data = pd.read_csv('AirPassengers.csv')
data['Month'] = data['Month'].apply(lambda dates: dt.datetime.strptime(dates, '%Y-%m'), 1)
data.index = data['Month']
data = data.drop('Month', 1)
print(data.shape)
data.head(3)

(144, 1)


Passengers
Month                 
1949-01-01         112
1949-02-01         118
1949-03-01         132

Всего у нас 144 строки и для теста оставим последние 7 записей

In [7]:
train = data.iloc[:130]
test = data.iloc[130:]
print(train.shape, test.shape)

(130, 1) (14, 1)


In [8]:
train.to_csv("train.csv")
test.to_csv("test.csv")